parquet/src/main/java/org/apache/iceberg/parquet/ParquetReader.java

```diff
@@ -79,9 +83,11 @@ private ReadConf<T> init() {
               nameMapping,
               reuseContainers,
               caseSensitive,
-              null);
-      this.conf = readConf.copy();
-      return readConf;
+              null)) {
+        this.conf = readConf.copy();
```

Comment:
The reason why `ReadConf` is not closed here was because we wanted to reuse the reader rather than opening a new one. This change will cause normal file reads to add an additional GET request round trip to open the file again.

What is the case where this wasn't being closed? I think that the originally logic should be safe because the `init` method is only called if a `FileIterator` is created, and that calls `ReadConf.reader()`. So either a new reader is opened, or the existing reader is reused and will be closed by the iterator.

Reply:
You call `this.conf = readConf.copy();`. `copy()` calls a private constructor that does not copy the reader itself: 

```
private ReadConf(ReadConf<T> toCopy) {
    this.reader = null;
```

so the reader is created to just fill all the other fields and then immediately discarded without getting closed.

then, inside the iterator you call `conf.reader()` which now creates a reader because it's null:

```
  ParquetFileReader reader() {
    if (reader != null) {
      reader.setRequestedSchema(projection);
      return reader;
    }
```

i suppose the other way to "fix" this would be to copy the reader as part of the `copy` but the fact that was done looked very deliberate, perhaps you have a context why.

Reply:
The temporary reader that doesn't get closed is basically used to read file schema.